From 70d15d7fa10157ba601774dc1aa7df4ff2bfd25f Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Fri, 9 Feb 2018 11:04:01 +0100
Subject: [PATCH 20/22] ME176C: pwm: crc: Turn off backlight when shutting down

We need to turn off the backlight when shutting down to power off
the screen correctly. Otherwise, pressing the power button after
shutdown will let the screen flash for a second.
---
 drivers/pwm/pwm-crc.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/pwm/pwm-crc.c b/drivers/pwm/pwm-crc.c
index bd0ebd04856a..6ee577cab321 100644
--- a/drivers/pwm/pwm-crc.c
+++ b/drivers/pwm/pwm-crc.c
@@ -132,9 +132,18 @@ static int crystalcove_pwm_remove(struct platform_device *pdev)
 	return pwmchip_remove(&pwm->chip);
 }
 
+static void crystalcove_pwm_shutdown(struct platform_device *pdev)
+{
+	struct crystalcove_pwm *pwm = platform_get_drvdata(pdev);
+
+	/* Turn off backlight */
+	regmap_write(pwm->regmap, BACKLIGHT_EN, 0);
+}
+
 static struct platform_driver crystalcove_pwm_driver = {
 	.probe = crystalcove_pwm_probe,
 	.remove = crystalcove_pwm_remove,
+	.shutdown = crystalcove_pwm_shutdown,
 	.driver = {
 		.name = "crystal_cove_pwm",
 	},
-- 
2.16.2

